### PR TITLE
LINK-2083 | Authentication for web store webhooks

### PR DESCRIPTION
**Pull request by harriris-vincit**
### Description
Adds authentication for Talpa webhooks.

**NOTE! To be merged in August 2024 before production testing & deployment.**
### Closes
[LINK-2083](https://helsinkisolutionoffice.atlassian.net/browse/LINK-2083)

[LINK-2083]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-2083?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr945.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr945.api.dev.hel.ninja** :rocket::rocket::rocket:

File: registrations/permissions.py [hunk 1]
```diff
@@ -217,3 +217,8 @@ def has_object_permission(
         return request.user.is_superuser or request.user.is_financial_admin_of(
             obj.publisher
         )
+
+
+class WebStoreWebhookPermission(permissions.BasePermission):
+    def has_permission(self, request: Request, view: APIView) -> bool:
+        return request.user.is_authenticated
```
Comment (tuomas777) [inline registrations/permissions.py:224]: There is [rest_framework.permissions.IsAuthenticated](https://www.django-rest-framework.org/api-guide/permissions/#isauthenticated) that does the same so I don't think this is needed.

Comment (harriris-vincit) [reply registrations/permissions.py]: Ah yes, the same permission class that I removed some time ago and apparently had my memory erased after that. Changed as suggested.

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr945.api.dev.hel.ninja** :rocket::rocket::rocket:

**Review by tuomas777 — APPROVED** [review]
LGTM 👍 🐱

**Pull request merged**